ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -161,7 +165,26 @@ private short getDefaultClusterReplication() {
   @Override
   public boolean isTopicExists(final String topic) {
     LOG.trace("Checking for existence of topic '{}'", topic);
-    return listTopicNames().contains(topic);
+    try {
+      ExecutorUtil.executeWithRetries(
+          () -> adminClient.get().describeTopics(
+              ImmutableList.of(topic),
+              new DescribeTopicsOptions().includeAuthorizedOperations(true)
+          ).values().get(topic).get(),
+          RetryBehaviour.ON_RETRYABLE
+      );
+      return true;
+    } catch (final Exception e) {
+      if (Throwables.getRootCause(e) instanceof UnknownTopicOrPartitionException) {
+        return false;
+      }
+
+      if (Throwables.getRootCause(e) instanceof TopicAuthorizationException) {
```

Comment:
I'm not sure I understand the reasoning for returning false here. We don't know if the topic exists or not at this point right?

Reply:
I'll double check @AlanConfluent's comment, but my intuition was that if you're not authorized to describe a topic it's not going to show up on list topics. This is usually what I've seen in other services, I get a 404 if I'm not using the right credentials, not a 403.

Reply:
After further inspection, I'll remove this check and just let it propegate. It looks like if you try to describe a topic you don't have permissions for it just throws the epxected `UnknownTopicOrPartitionException`